app/backend/src/couchers/servicers/communities.py

```diff
@@ -107,19 +104,25 @@ def GetCommunity(self, request, context):
 
     def ListCommunities(self, request, context):
         with session_scope() as session:
-            page_size = min(MAX_PAGINATION_LENGTH, request.page_size or MAX_PAGINATION_LENGTH)
-            next_node_id = int(request.page_token) if request.page_token else 0
+            page_size = request.page_size if request.page_size else MAX_PAGINATION_LENGTH
+            next_node_name = request.page_token if request.page_token else ""
+
             nodes = (
                 session.query(Node)
+                .select_from(Cluster)
+                .join(Node, Cluster.parent_node_id == Node.id)
                 .filter(or_(Node.parent_node_id == request.community_id, request.community_id == 0))
-                .filter(Node.id >= next_node_id)
-                .order_by(Node.id)
+                .filter(Cluster.name >= next_node_name)
+                .filter(Cluster.is_official_cluster == True)
+                .order_by(Cluster.name)
                 .limit(page_size + 1)
                 .all()
             )
+
+            communities = [community_to_pb(node, context) for node in nodes]
             return communities_pb2.ListCommunitiesRes(
-                communities=[community_to_pb(node, context) for node in nodes[:page_size]],
-                next_page_token=str(nodes[-1].id) if len(nodes) > page_size else None,
+                communities=communities[:page_size],
+                next_page_token=str(communities[-1].name) if len(nodes) > page_size else None,
```

Comment:
You're turning all communities into the proto format, including the last one, which is just there to figure out if we need to fetch more. It's also not good to use `communities[-1].name` for the name: that's an implementation detail of the proto format, and you should just use the `cluster.name`.

With the snippet I had before, you could do `communities=[community_to_pb(node, context) for node, _ in node_and_clusters[:page_size]]` (assuming you rename `nodes` to `node_and_clusters`) and then `node_and_clusters[-1][1].name if ...` in the second bit (also the cast to `str` is unnecessary since it's already a string).